### PR TITLE
use most recent namespacemetadata in x-repo search indexing

### PR DESCRIPTION
**Pull request by jctanner**
https://github.com/pulp/pulp_ansible/issues/1494

### CODE REVIEW

File: pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py:1481]: I'm still unsure whether that order is stable.
(Reminds me that i want tasks to be able to actually return something.)

File: pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py:1503]: This is the same as kwargs above, no?
Can we call it a little more useful and reuse it? dist_kwargs, maybe?

File: pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py:1522]: Can we not assert on that? Wouldn't we even know the sha256 of the image we generated above?

**Review by mdellweg — APPROVED** [review]
I'm not enough of an expert for these namespaces, but the test looks well written.
Please wait for a second review.

File: pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py
Comment (jctanner) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py:1522]: It another bug that I found last night. I'm not sure if it's an issue in the serializer or what, but I wasn't sure if it's something we can easily solve in this PR and still have a backportable patch. We're most interested in the avatar_url for the UI elements and the sha256 is less concerning.

File: pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py
Comment (jctanner) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py:1481]: I tried to use gen_object_with_cleanup, but something about that api client or maybe the async task is incompatible.

Comment (mdellweg) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py]: Hah, no a task with two returned entitites is not compatible with that fixture.

Comment (mdellweg) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py]: You can still use the add_to_cleanup fixture though.

File: pulp_ansible/app/tasks/collectionversion_index.py
Comment (awcrosby) [inline pulp_ansible/app/tasks/collectionversion_index.py:162]: this looks like it gets all `AnsibleNamespaceMetadata` objects, not just the ones scoped to the repository_version

Comment (jctanner) [reply pulp_ansible/app/tasks/collectionversion_index.py]: fixed

File: pulp_ansible/app/tasks/collectionversion_index.py
Comment (awcrosby) [inline pulp_ansible/app/tasks/collectionversion_index.py:190]: I'm not sure what circumstance would make `changed_collections` be `None`, but  (prior to this PR and also with this PR) this logic seems to do this:

if `changed_collections` is `None`, it will not `continue`, so it will attempt the rest of the create|update index... is that desired?

Comment (jctanner) [reply pulp_ansible/app/tasks/collectionversion_index.py:190]: That is correct. However changed_collections reflects changes to CVs, deprecatation, signatures and now namespacemetadata.

Comment (awcrosby) [reply pulp_ansible/app/tasks/collectionversion_index.py:190]:
```suggestion
        if changed_collections is None:
            continue
        if (
            (namespace, name) not in changed_collections
            and (namespace, "*") not in changed_collections
        ):
```

IIUC, and this is logically correct, this is a suggestion to be explicty that there is no indexing when no changed collections, optional edit

Comment (awcrosby) [reply pulp_ansible/app/tasks/collectionversion_index.py:190]: probably better to just remove check for `changed_collections is not None`

**Review by rochacbruno — APPROVED** [review]

**Pull request merged**
